zmon_aws_agent/postgresql.py

```diff
@@ -58,3 +59,119 @@ def get_databases_from_clusters(pgclusters, infrastructure_account, region,
         logger.exception("Failed to make Database entities for PostgreSQL clusters!")
 
     return entities
+
+
+def collect_eip_addresses(infrastructure_account):
+    ec2 = boto3.client('ec2')
+
+    addresses = call_and_retry(ec2.describe_addresses)['Addresses']
+
+    return [a for a in addresses if a['NetworkInterfaceOwnerId'] == infrastructure_account.split(':')[1]]
+# FIXME: depend on region, too?
+
+
+def filter_asgs(infrastructure_account, asgs):
+    return [gr for gr in asgs
+            if gr['infrastructure_account'] == infrastructure_account
+            and 'spilo_cluster' in gr.keys()]
+
+
+def filter_instances(infrastructure_account, instances):
+    return [i for i in instances if i['infrastructure_account'] == infrastructure_account]
+
+
+def collect_launch_configurations(infrastructure_account):
+    asg = boto3.client('autoscaling')
+    lc_paginator = asg.get_paginator('describe_launch_configurations')
+    lcs = lc_paginator.paginate().build_full_result()['LaunchConfigurations']
+
+    user_data = {}
+
+    for lc in lcs:
+        # LaunchConfigurationName takes the form of spilo-your-cluster-AppServerInstanceProfile-66CCXX77EEPP
+        lc_name = '-'.join(lc['LaunchConfigurationName'].split('-')[1:-2])
+        user_data[lc_name] = lc['UserData']
+
+    return user_data
+
+
+def extract_eipalloc_from_lc(launch_configuration, cluster_name):
+    import yaml
+    import base64
+
+    lc = launch_configuration.get(cluster_name, '')
+
+    user_data = base64.decodebytes(lc.encode('utf-8')).decode('utf-8')
+    user_data = yaml.safe_load(user_data)
+
+    return user_data['environment'].get('EIP_ALLOCATION', '')
+
+
+def get_postgresql_clusters(region, infrastructure_account, asgs, insts):
```

Comment:
Is this needed on all agents? can we have an empty list if not needed? Env variable could help here.

Reply:
Most possibly not, but we cannot know easily if yes or no.  Unless you know about a reliable way of finding out if there are Spilo clusters in an account (I cannot currently think of one).  And in the end, this returns exactly the empty list, if there are no Spilo ASGs.